x/nexus/keeper/lockable_asset.go

```diff
@@ -31,14 +31,14 @@ type lockableAsset struct {
 func newLockableAsset(ctx sdk.Context, nexus types.Nexus, ibc types.IBCKeeper, bank types.BankKeeper, coin sdk.Coin) (lockableAsset, error) {
```

Comment:
Previously, the inputs would be always be bank Coin denoms, but now we're allowing both representations as inputs due to our cross-chain transfer logic
```suggestion
// The coin denom can either be an actual bank Coin (e.g. uaxl, ICS20 coin) or the registered asset name (e.g. base denom for an ICS20 coin)
func newLockableAsset(ctx sdk.Context, nexus types.Nexus, ibc types.IBCKeeper, bank types.BankKeeper, coin sdk.Coin) (lockableAsset, error) {
```

Reply:
I would also support creating two separate constructors `newLockableAsset` and `newLockableAssetFromCoin` to distinguish the two input representations, making it more clear what type of constructor is needed in which place. A catch-all constructor seems a bit more prone to weird edge cases

Reply:
I am not a fan of two separate constructors. The words coin and asset are confusing enough that no one understands the difference after a while.